runtime/doc/gui.txt

```diff
@@ -66,6 +66,19 @@ Stop or detach the current UI
 
                 Note: Not supported on Windows, currently.
 
+------------------------------------------------------------------------------
+Restart the embedded Nvim server
+
+                                                *:restart*
+:restart
+                Detaches the embedded server from the UI and then restarts
+                it. This fails when changes have been made
+                and Vim refuses to |abandon| the current buffer.
+
+:restart!
```

Comment:
off-topic / non-blocker:: imo this is an uninteresting use of "!" (although common). Users can always do `:wall` to save everything before restart. And `:confirm restart` to opt-in to confirm "unsaved changes".

We might want to use "!" to mean "don't restore state/session" or something like that.

Don't need to figure this out right now.